Automattic/wp-calypso
# Marketplace Reviews: Show pending review warning

Fixes https://github.com/Automattic/dotcom-forge/issues/4824

## Proposed Changes

* Only display the write a review card if the user doesn't have a review (not even pending)
* On the Reviews list, show the user review even if its pending
  * Put a warning on the top when pending

![CleanShot 2024-01-03 at 13 31 33@2x](https://github.com/Automattic/wp-calypso/assets/5039531/50d29b6a-b53a-4001-a7c0-6b170d1df455)


## Testing Instructions

* Add a review on pending state on a plugin. You can use `/marketplace/test/:site` to add a review.
* Go to the plugin details page
* Confirm the review is not displayed on the cards
* Click on `Read all reviews`
* Check if your review is displayed but with a warning added above.
* Enable the flag `"marketplace-reviews-notification"` and check the message  is appended with `You will be notified once it is published.`. This should be released as soon as this feature is ready.

## Pre-merge Checklist
